cmd/cstor-volume-mgmt/controller/volume-controller/after_queue_handler.go

```diff
@@ -0,0 +1,175 @@
+		// CheckValidVolume is to check if volume attributes are correct.
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return common.StatusOffline, err
+		}
+
+		err = volume.CreateVolume(cStorVolumeGot)
+		if err != nil {
+			return "", err
+		}
+		break
+
+	case "modify":
+		glog.Info("modify event") //ignoring as of now
+
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return "", err
+		}
+
+		err = volume.CreateVolume(cStorVolumeGot)
+		if err != nil {
+			return "", err
+		}
+		break
+
+	case "destroy":
+		glog.Info("destroy event")
+		return common.StatusIgnore, nil
+	}
+
+	return common.StatusIgnore, nil
+}
+
+// enqueueCstorVolume takes a CStorVolume resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than CStorVolumes.
+func (c *CStorVolumeController) enqueueCStorVolume(obj interface{}, q common.QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.Key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+// getVolumeResource returns object corresponding to the resource key
+func (c *CStorVolumeController) getVolumeResource(key string) (*apis.CStorVolume, error) {
+	// Convert the key(namespace/name) string into a distinct name
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil, nil
+	}
+
+	cStorVolumeGot, err := c.clientset.OpenebsV1alpha1().CStorVolumes().Get(name, metav1.GetOptions{})
+	if err != nil {
+		// The cStorVolume resource may no longer exist, in which case we stop
+		// processing.
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("cStorVolumeGot '%s' in work queue no longer exists", key))
+			return nil, nil
+		}
+
+		return nil, err
+	}
+	return cStorVolumeGot, nil
+}
+
+// removeFinalizer is to remove finalizer of cstorvolume resource.
+func (c *CStorVolumeController) removeFinalizer(cStorVolumeGot *apis.CStorVolume) error {
+	if len(cStorVolumeGot.Finalizers) > 0 {
+		cStorVolumeGot.Finalizers = []string{}
+	}
+	_, err := c.clientset.OpenebsV1alpha1().CStorVolumes().Update(cStorVolumeGot)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// IsRightCStorVolumeMgmt is to check if the volume request is for particular pod/application.
+func IsRightCStorVolumeMgmt(cStorVolume *apis.CStorVolume) bool {
+	if os.Getenv("cstorid") == string(cStorVolume.ObjectMeta.UID) {
+		glog.Infof("right sidecar")
+		return true
+	}
+	glog.Infof("wrong sidecar")
+	return false
+}
+
+// IsDestroyEvent is to check if the call is for cStorVolume destroy.
+func IsDestroyEvent(cStorVolume *apis.CStorVolume) bool {
+	if cStorVolume.ObjectMeta.DeletionTimestamp != nil {
+		glog.Infof("cstor destroy event")
+		return true
+	}
+	glog.Infof("cstor modify event")
```

Comment:
same comments !!!
Were we using these logs to debug during development.
Lets review our code once again if above is true.